### PR TITLE
Fix styling and move towards standard TemplateResponse

### PR DESCRIPTION
**Pull request by skjnldsv**
- [ ] Requires new RENDER_AS_CLEAN template on server that provide bare minimum styling (variables only) and core js

### CODE REVIEW

File: src/workersrc.js [hunk 1]
```diff
@@ -24,11 +24,12 @@
 
 import canDownload from './utils/canDownload'
 import redirectIfNotIframe from './utils/redirectIfNotIframe'
+import { loadState } from '@nextcloud/initial-state'
 
 /**
  * Checks if the page is displayed in an iframe. If not redirect to /.
  **/
-redirectIfNotIframe()
+// redirectIfNotIframe()
```
Comment (skjnldsv) [inline src/workersrc.js:32]: TODO: revert before merge, this allows to display the iframe without being redirected to the nc dashboard

**Comment by jancborchardt** [comment]
For the record: How it looks like now, and what we should fix:
- [ ] Viewer should span whole width and also go down to the bottom
- [ ] Prevent styling of core to leak here, as you see from the input fields and selects. Or we increase the height of the action bar and adjust that.
- [ ] Possibly we still need access to variables, but maybe not?
- [ ] Background could be fully transparent as we already have the modal background
- [ ] The actions bar should not have any gradient nor this background-image texture – maybe it could also be transparent

image

**Comment by szaimen** [comment]
Concerning the style of pdf.js, there was a redesign just merged here mozilla/pdf.js#11077

Maybe this change get merged into the pdfviewer first and then see what is still broken design wise?

**Comment by skjnldsv** [comment]
> Concerning the style of pdf.js, there was a redesign just merged here [mozilla/pdf.js#11077](https://github.com/mozilla/pdf.js/pull/11077)

Once it's released, sure!

**Comment by szaimen** [comment]
> Once it's released, sure!

[v2.6.347](https://github.com/mozilla/pdf.js/releases/tag/v2.6.347) of pdf.js was just released 🎉

**Comment by skjnldsv** [comment]
> [v2.6.347](https://github.com/mozilla/pdf.js/releases/tag/v2.6.347) of pdf.js was just released

as alpha

**Comment by jancborchardt** [comment]
For reference, this is the small style modifications I experimented with – let’s see if they still work when we update pdf.js:
```
// Transparent background
body {
	background-color: transparent;
	background-image: none;
}

// Semi-transparent background for toolbars
#toolbarContainer,
.findbar,
.secondaryToolbar,
#toolbarSidebar {
	background-color: rgba(0, 0, 0, 0.85);
	background-image: none;
}
```

**Comment by szaimen** [comment]
> For reference, this is the small style modifications I experimented with – let’s see if they still work when we update pdf.js:
> 
> ```
> // Transparent background
> body {
> 	background-color: transparent;
> 	background-image: none;
> }
> 
> // Semi-transparent background for toolbars
> #toolbarContainer,
> .findbar,
> .secondaryToolbar,
> #toolbarSidebar {
> 	background-color: rgba(0, 0, 0, 0.85);
> 	background-image: none;
> }
> ```

I am eager to see if this still works now that #290 is merged :)

**Comment by skjnldsv** [comment]
Yep, @jancborchardt you can double check with latest master of files_pdfviewer. there is additional design changes :)

**Comment by susnux** [comment]
@skjnldsv are you still working on this or should we close this?

**Pull request closed without merging**